packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -640,6 +656,14 @@ function isCouponApplied( { coupon_savings_integer = 0 }: ResponseCartProduct )
 	return coupon_savings_integer > 0;
 }
 
+function hasPartnerCoupon( coupon: string, products?: ResponseCartProduct[] ): boolean {
+	const productHasSublabel =
+		products && products.find( ( product: ResponseCartProduct ) => getSublabel( product ) );
+	const isPartnerCoupon = coupon.startsWith( 'IONOS_' );
+
+	return ( productHasSublabel && isPartnerCoupon ) || false;
+}
+
```

Comment:
Helper that tells whether or not to render the _PartnerLogo_ contents. Currently being used inside the _PartnerLogo_ component, which might seem redundant to have two checks for rendering the same content in different places, the first being at _WPNonProductLineItem_ and the second being inside _PartnerLogo_ component, but in the future this conditionals might scale to support more complex cases.

Reply:
The logic for defining _**productHasSubLabel**_ doesn't seem consistent. But according to https://github.com/Automattic/wp-calypso/pull/54811 , Jetpack checkout will only have a single product.

Reply:
Can you explain more about why this logic needs to look at all products in the cart (at least for `PartnerLogo`) for a sublabel and not just the product being rendered?